apps/meteor/client/lib/rooms/roomTypes/public.ts

```diff
@@ -87,6 +87,9 @@ roomCoordinator.add(PublicRoomType, {
 	},
 
 	findRoom(identifier) {
+		if (!hasPermission('view-c-room')) {
+			return;
+		}
```

Comment:
```suggestion
```

It's better not to have this test at all. It'll only cause the room data to be loaded from the server every time.